src/Sarif/EnumeratedArtifact.cs

```diff
@@ -19,131 +19,78 @@ public EnumeratedArtifact(IFileSystem fileSystem)
         internal byte[] bytes;
         internal string contents;
 
-        private Encoding encoding;
-
         public Uri Uri { get; set; }
 
-        public bool IsBinary
-        {
-            get
-            {
-                GetArtifactData();
-                return this.contents == null;
-            }
-        }
-
         public Stream Stream { get; set; }
 
-        public Encoding Encoding
-        {
-            get
-            {
-                if (encoding == null)
-                {
-                    GetArtifactData();
-                }
-                return this.encoding;
-            }
-
-            set => this.encoding = value;
-        }
+        public Encoding Encoding { get; set; }
 
         internal IFileSystem FileSystem { get; set; }
 
+        public bool IsBinary { get; set; }
+
         public string Contents
         {
-            get => GetArtifactData().text;
+            get => !IsBinary? GetContents() : throw new InvalidOperationException();
             set => this.contents = value;
         }
 
         public byte[] Bytes
         {
-            get => GetArtifactData().bytes;
+            get => IsBinary ? GetBytes() : throw new InvalidOperationException();
             set => this.bytes = value;
         }
 
-        private (string text, byte[] bytes) GetArtifactData()
+        private string GetContents()
         {
-            if (this.contents != null)
-            {
-                return (this.contents, bytes: null);
-            }
+            if (this.contents != null) { return this.contents; }
 
-            if (this.bytes != null)
+            if (Stream == null && this.contents == null)
             {
-                return (text: null, this.bytes);
-            }
-
-            if (Stream == null && this.contents == null && this.bytes == null)
-            {
-                if (Uri == null ||
-                    !Uri.IsAbsoluteUri ||
-                    (Uri.IsAbsoluteUri && !Uri.IsFile))
-                {
-                    throw new InvalidOperationException("An absolute URI pointing to a file location was not available.");
-                }
-
-                // This is our client-side, disk-based file retrieval case.
-                this.Stream = FileSystem.FileOpenRead(Uri.LocalPath);
-            }
+                // TBD we actually have no validation URI is non-null yet.
+                contents = Uri!.IsFile
```

Comment:
>IsFile [](http://example.com/codeflow?start=32&length=6)

not sure if applies here,
Uri.IsFile will throws an InvalidOperationException when it is a relative Uri